rosdep/python.yaml

```diff
@@ -2564,19 +2564,15 @@ python-mock:
     xenial_python3: [python3-mock]
     yakkety: [python-mock]
     zesty: [python-mock]
-python-more-itertools-pip:
```

Comment:
@tfoote had some arguments against that: https://github.com/ros/rosdistro/pull/19058#issuecomment-789358134

> Maintaining parallel rules with -pip and regular leads to many more problems because the pip version will mask the debian based one. This means that if any package on the system asks for the pip version all packages on the system will get the pip version, even if they asked for the apt version.

I don't have an opinion about this since we simply don't have packages with the `-pip` prefix.

So I will let you two come to a consensus and implement that :+1:

Reply:
Sorry I wasn't specific in my description. I meant sunsetting as a planned phase out. We should keep the old key, but have it not resolve on the newer platforms on which the apt key is available. To that end we should call out for Ubuntu xenial version for the pip key and not define any other versions as it's the only supported version of ubuntu still available. This will only force users of bionic or newer to switch over as they would have the conflicting definitions.